### PR TITLE
Check: default model and provider are required

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Check: default model and provider are required

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Configuration Update
- [ ] Bump-up dependent library

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Comment by syedriko** [comment]
@tisnik We could also make https://github.com/openshift/lightspeed-service/blob/e9f21d40e72d1c468bfe4b43a53ad640a1f2f480/ols/app/models/config.py#L825 less generic and just code it around default provider and model.

**Comment by tisnik** [comment]
> @tisnik Thanks for fixing this, it's a leftover from when we had multiple model roles and it was ok not to have them set. Speaking of which, could you remove
> 
> https://github.com/openshift/lightspeed-service/blob/e9f21d40e72d1c468bfe4b43a53ad640a1f2f480/ols/constants.py#L197
> and the loop at
> 
> https://github.com/openshift/lightspeed-service/blob/e9f21d40e72d1c468bfe4b43a53ad640a1f2f480/tests/unit/utils/test_config.py#L170
> ?

removed

**Comment by tisnik** [comment]
> @tisnik We could also make
> 
> https://github.com/openshift/lightspeed-service/blob/e9f21d40e72d1c468bfe4b43a53ad640a1f2f480/ols/app/models/config.py#L825
> less generic and just code it around default provider and model.

updated accordingly

File: ols/app/models/config.py [hunk 1]
```diff
@@ -822,37 +822,31 @@ def __eq__(self, other: object) -> bool:
...
+        model_specified = selected_default_model is not None
+
+        if not provider_specified:
             raise InvalidConfigurationError(
-                f"{provider_attr_name} is specified, but {model_attr_name} is missing"
+                "default_provider with provider name is missing"
             )
-        elif not provider_specified and model_specified:
+        if not model_specified:
+            raise InvalidConfigurationError("default_model with model name is missing")
```
Comment (syedriko) [inline ols/app/models/config.py:837]: should we just say "default_provider is missing" and "default_model is missing"?

File: tests/unit/utils/test_config.py [hunk 2]
```diff
@@ -167,101 +166,6 @@ def test_invalid_config():
...
-    type: memory
-    memory:
-      max_entries: 1000
-  {role}_model: m1
-    """,
-            InvalidConfigurationError,
-            f"{role}_model is specified, but {role}_provider is missing",
-        )
-
     check_expected_exception(
```
Comment (syedriko) [inline tests/unit/utils/test_config.py:169]: I didn't mean to remove these tests completely, just remove the loop and hadcode the role parameter to "default"

File: ols/app/models/config.py (hunk 1 above)
Comment (tisnik) [reply ols/app/models/config.py]: updated

File: tests/unit/utils/test_config.py (hunk 2 above)
Comment (tisnik) [reply tests/unit/utils/test_config.py:169]: ah sure thing, sorry for misunderstanding. updated

**Pull request merged**